Framework/DataHandling/inc/MantidDataHandling/LoadNGEM.h

```diff
@@ -14,14 +14,18 @@ namespace Mantid {
 namespace DataHandling {
 
 static constexpr uint64_t CONTIN_ID_VALUE = 0x4F;
+static constexpr uint64_t EVENT_ID_MASK = 0x40;
 
 /// Generic event to separate bits.
 struct GenericEvent {
   uint64_t t0id : 24;      // T0 ID
   uint64_t reserved2 : 32; // Reserved for non-generics
   uint64_t contin : 8;     // 0x4F Continuation Code
   uint64_t reserved1 : 56; // Reserved for non-generics
-  uint64_t id : 8;         // 0x4E Event ID
+  uint64_t id : 8;         // Event ID
+  bool check() const {
+    return (id & EVENT_ID_MASK) != 0 && contin == CONTIN_ID_VALUE;
```

Comment:
It is checking for an `id` of the form `0x4?` that seems to be the reserved space used by nGEM for its ids (but I haven't seen this categorically in writing). At the moment this would include all events we handle plus some that are in the manual but we don't see (because we don't run in those modes). So the idea was to be able to flag up a "potentially correct but don't yet handle" data from "this is just corrupt".

Reply:
The test does rely on `id` being only 8 bits, 0x140 would also pass for example, but `id` cannot exceed 0xff so the situation does not occur. I can add a comment about this.

Reply:
Actually, it's less robust than I though - I'll push a change in a bit

Reply:
Probably id >> 6 & ~1 == 0

Reply:
but may simplify!